react/features/base/tracks/functions.js

```diff
@@ -295,6 +296,33 @@ export function getLocalAudioTrack(tracks) {
     return getLocalTrack(tracks, MEDIA_TYPE.AUDIO);
 }
 
+/**
+ * Returns the local desktop track.
+ *
+ * @param {Track[]} tracks - List of all tracks.
+ * @param {boolean} [includePending] - Indicates whether a local track is to be returned if it is still pending.
+ * A local track is pending if {@code getUserMedia} is still executing to create it and, consequently, its
+ * {@code jitsiTrack} property is {@code undefined}. By default a pending local track is not returned.
+ * @returns {(Track|undefined)}
+ */
+export function getLocalDesktopTrack(tracks, includePending = false) {
+    return (
+        getLocalTracks(tracks, includePending)
+            .find(t => t.mediaType === MEDIA_TYPE.SCREENSHARE || t.videoType === VIDEO_TYPE.DESKTOP));
```

Comment:
why is the `||` logic needed? is it to make it work with both new and legacy flow?

Reply:
Yes, this would make it work for both the new and the legacy flow.